### PR TITLE
Convert Virtual Machine pages to Laravel

### PR DESCRIPTION
**Pull request by Jellyfrog**
PowerState class is meant to be generic, not only for virtual machines.

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: app/Models/Vminfo.php [hunk 1]
```diff
@@ -3,11 +3,28 @@
...
+use LibreNMS\Util\Rewrite;
 
 class Vminfo extends DeviceRelatedModel
 {
     use HasFactory;
 
     protected $table = 'vminfo';
     public $timestamps = false;
+
+    public function vmDevice()
```
Comment (Jellyfrog) [inline app/Models/Vminfo.php:16]: Need a better name here maybe

File: LibreNMS/Enum/PowerState.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        'running' => self::ON,
+
+        'suspended' => self::SUSPENDED,
+        'paused' => self::SUSPENDED,
+    ];
+
+    /**
+     * @param int|string $state
+     */
+    public static function stateLabel($state): array
```
Comment (Jellyfrog) [inline LibreNMS/Enum/PowerState.php:43]: Should we do these functions in our Enum classes? I dont know a better place.

Comment (murrant) [reply LibreNMS/Enum/PowerState.php:43]: I think maybe in the Html utility class or something.  The enum does feel like an odd place for it.

**Review by murrant — APPROVED** [review]
LGTM

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-69-release-changelog-november-2020/14124/1

**Pull request merged**